apps/veil/src/pages/explore/ui/pair-card.tsx

```diff
@@ -3,170 +3,136 @@ import { Star, CandlestickChart } from 'lucide-react';
 import { Button } from '@penumbra-zone/ui/Button';
 import { Text } from '@penumbra-zone/ui/Text';
 import { Density } from '@penumbra-zone/ui/Density';
 import { AssetIcon } from '@penumbra-zone/ui/AssetIcon';
-
-import { SummaryData } from '@/shared/api/server/summary/types';
-import { Skeleton } from '@/shared/ui/skeleton';
-import SparklineChart from './sparkline-chart.svg';
 import ChevronDown from './chevron-down.svg';
 import { PreviewChart } from './preview-chart';
-import { useRegistryAssets } from '@/shared/api/registry';
-
-const ShimmeringBars = () => {
-  return (
-    <>
-      <div className='w-16 h-4 my-1'>
-        <Skeleton />
-      </div>
-      <div className='w-10 h-4'>
-        <Skeleton />
-      </div>
-    </>
-  );
-};
+import { SummaryWithPrices } from '@/shared/api/server/summary';
+import { useGetMetadata } from '@/shared/api/assets';
+import { toValueView } from '@/shared/utils/value-view';
+import { convertPriceToDisplay } from '@/shared/math/price';
 
-const getTextSign = (summary: SummaryData): ReactNode => {
-  if (summary.change.sign === 'positive') {
+const getTextSign = (change: number): ReactNode => {
+  if (change > 0) {
     return <ChevronDown className='size-3 rotate-180 inline-block' />;
   }
-  if (summary.change.sign === 'negative') {
+  if (change < 0) {
     return <ChevronDown className='size-3 inline-block' />;
   }
   return null;
 };
 
-const getColor = (summary: SummaryData): string => {
-  if (summary.change.sign === 'positive') {
+const getColor = (change: number): string => {
+  if (change > 0) {
     return 'text-success-light';
   }
-  if (summary.change.sign === 'negative') {
+  if (change < 0) {
     return 'text-destructive-light';
   }
   return 'text-neutral-light';
 };
 
-export type PairCardProps =
-  | {
-      loading: true;
-      summary: undefined;
-    }
-  | {
-      loading: false;
-      summary: SummaryData;
-    };
+export interface PairCardProps {
+  summary: SummaryWithPrices;
+}
 
-export const PairCard = ({ loading, summary }: PairCardProps) => {
+export const PairCard = ({ summary }: PairCardProps) => {
   const today = new Date();
   const yesterday = subDays(new Date(), 1);
 
-  const { data: assets } = useRegistryAssets();
-  const usdcMetadata = assets.find(asset => asset.symbol === 'USDC');
+  const getMetadata = useGetMetadata();
+  const startMetadata = getMetadata(summary.start);
+  if (!startMetadata) {
+    throw new Error(`unknown asset: ${summary.start.toJsonString()}`);
+  }
+  const endMetadata = getMetadata(summary.end);
+  if (!endMetadata) {
+    throw new Error(`unknown asset: ${summary.end.toJsonString()}`);
+  }
+  const liquidityMetadata = getMetadata(summary.liquidity.assetId);
+  const volumeMetadata = getMetadata(summary.volume.assetId);
 
   return (
     <Link
-      href={loading ? `/trade` : `/trade/${summary.baseAsset.symbol}/${summary.quoteAsset.symbol}`}
+      href={`/trade/${startMetadata.symbol}/${endMetadata.symbol}`}
```

Comment:
suggestion (non-blocking): use [`getTradePairPath` function](https://github.com/penumbra-zone/web/blob/main/apps/veil/src/shared/const/pages.ts#L34) to get the trade link